apps/etl-func/src/adapters/cosmos/event-collector.ts

```diff
@@ -1,63 +1,56 @@
-import {
-  EventCollector,
-  EventsSummary,
-  eventsSummarySchema,
-} from "@/domain/event.js";
+import { EventCollector, EventsSummary } from "@/domain/event.js";
 import { TelemetryEventName, TelemetryService } from "@/domain/telemetry.js";
 import { Container, ItemResponse } from "@azure/cosmos";
 
 /**
- * Adapter for a cosmos weekly event collector.
- * The goal is to store the number of events sent to the event hub
- *  into a container with year as partition key and week number as model id.
+ * Adapter for a cosmos event collector.
+ * The goal is to store the number of events
+ * into a container using the year as partition key.
  * */
-export class CosmosWeeklyEventCollector<T> implements EventCollector<T> {
+export class CosmosSummaryCollector<T extends { id: string }>
+  implements EventCollector<T>
+{
   #container: Container;
   #telemetry: TelemetryService;
 
   /**
    * @param container The name of the container where to store weekly events.
+   * @param telemetry The telemetry service to use
    * */
   constructor(container: Container, telemetry: TelemetryService) {
     this.#container = container;
     this.#telemetry = telemetry;
   }
 
-  private createSummary(events: T[]): EventsSummary {
+  /**
+   * Accept a batch of events and return the summary.
+   * */
+  private createSummary(events: [T, ...T[]]): EventsSummary {
     return {
       count: events.length,
-      id: this.getCurrentModelId(),
+      id: this.getBatchId(events),
       year: this.getCurrentPartitionKey(),
     };
   }
```

Comment:
why is this a method? we are creating a unnecessary coupling here

Reply:
[5a6d7ad](https://github.com/pagopa/io-messages/pull/130/commits/5a6d7ad7fb8475827c91e99444f8ddd2ccdfc48b)